
# HG changeset patch
# User Todd C. Miller <Todd.Miller@courtesan.com>
# Date 1437163106 21600
# Node ID 0fa324a7bb5665aec9d39cdf011eaed32db36685
# Parent  e4c7cda464759b555e46ad0de229f302ac22b112
Fix typo in sudo_sss_attrcpy() that caused a memory allocation error.

diff -r e4c7cda46475 -r 0fa324a7bb56 plugins/sudoers/sssd.c
--- a/plugins/sudoers/sssd.c	Wed Jul 15 12:36:02 2015 -0600
+++ b/plugins/sudoers/sssd.c	Fri Jul 17 13:58:26 2015 -0600
@@ -142,7 +142,7 @@
     sudo_debug_printf(SUDO_DEBUG_INFO, "malloc: cnt=%d", src->num_values);
 
     dst->name = strdup(src->name);
-    dst->values = reallocarray(NULL, dst->num_values, sizeof(char *));
+    dst->values = reallocarray(NULL, src->num_values, sizeof(char *));
     if (dst->name == NULL || dst->values == NULL)
 	goto oom;
     dst->num_values = src->num_values;
@@ -195,7 +195,6 @@
 
     for (i = 0; i < src->num_attrs; ++i) {
 	if (!sudo_sss_attrcpy(dst->attrs + i, src->attrs + i)) {
-	    sudo_warnx(U_("%s: %s"), __func__, U_("unable to allocate memory"));
 	    dst->num_attrs = i;
 	    sudo_sss_rulefree(dst);
 	    debug_return_bool(false);

